staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -300,14 +302,22 @@ func (flags *ApplyFlags) ToOptions(f cmdutil.Factory, cmd *cobra.Command, baseNa
 
 	var applySet *ApplySet
 	if flags.ApplySetRef != "" {
-		var applySetNs string
+		parent, err := ParseApplySetParentRef(flags.ApplySetRef, mapper)
+		if err != nil {
+			return nil, fmt.Errorf("invalid parent reference %q: %w", flags.ApplySetRef, err)
+		}
 		// ApplySet uses the namespace value from the flag, but not from the kubeconfig or defaults
-		if enforceNamespace {
-			applySetNs = namespace
+		// This means the namespace flag is required when using a namespaced parent.
+		if enforceNamespace && parent.IsNamespaced() {
+			parent.Namespace = namespace
 		}
-		if applySet, err = NewApplySet(flags.ApplySetRef, applySetNs, mapper); err != nil {
-			return nil, err
+		// TODO: is version.Get() the right thing? Does it work for non-kubectl package consumers?
+		tooling := ApplySetTooling{name: baseName, version: version.Get().String()}
```

Comment:
@soltysh this TODO is really a question for you: is what I'm doing here going to work for tools like oc?

Reply:
hmm, interesting. I see that `baseName` is injected as part of the signature of `NewCommandApply`. I don't really want to modify that for an alpha... I could expose a global variable OC could override, or perhaps I could do a `if basename == "kubectl"` check and have other tools be versionless for now. Wdyt?